contrib/win32/win32compat/shell-host.c

```diff
@@ -663,53 +790,41 @@ ProcessEvent(void *p)
 		wX = LOWORD(idObject);
 		wY = HIWORD(idObject);
 		
-		SMALL_RECT readRect;
 		readRect.Top = wY;
 		readRect.Bottom = wY;
 		readRect.Left = wX;
-		readRect.Right = ConSRWidth();
+		readRect.Right = wX;
 		
-		/* Set cursor location based on the reported location from the message */
-		CalculateAndSetCursor(pipe_out, wX, wY);
-		
-		COORD coordBufSize;
-		coordBufSize.Y = readRect.Bottom - readRect.Top + 1;
-		coordBufSize.X = readRect.Right - readRect.Left + 1;
+		coordBufSize.Y = 1;
+		coordBufSize.X = 1;
+
+		if (!IS_TT_VTX(termType) || wX < currentCol || wY < currentLine) {
+			/* send full line */
+			readRect.Right = ConSRWidth();
+			coordBufSize.Y += readRect.Bottom - readRect.Top;
+			coordBufSize.X += readRect.Right - readRect.Left;
+		}
+		bufferSize = coordBufSize.X * coordBufSize.Y;
+		if (bufferSize > MAX_EXPECTED_BUFFER_SIZE)
```

Comment:
Value of **coordBufSize.Y** always contain **1**, because Win32 console event **EVENT_CONSOLE_UPDATE_REGION** always contain only one line.